### PR TITLE
FINERACT-1932: Fineract modularization - fineract-savings

### PR DESCRIPTION
**Pull request by josehernandezfintecheandomx**
## Description

Create the initial fineract-savings module - used to savings products and accounts module classes and interfaces

[FINERACT-1932](https://issues.apache.org/jira/browse/FINERACT-1932)

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/interoperation/data/InteropTransactionData.java [hunk 1]
```diff
@@ -118,6 +118,7 @@ public static InteropTransactionData build(SavingsAccountTransaction transaction
 
         StringBuilder sb = new StringBuilder();
         int currLength = 0;
+        /*
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/interoperation/data/InteropTransactionData.java]: This looks a breaking change to me

**Comment by adamsaghy** [comment]
@josehernandezfintecheandomx Please find a way to fetch the Notes into the InterOpData as without it, it is a breaking change.

Please also check the spotless failures!

File: fineract-provider/src/main/java/org/apache/fineract/interoperation/data/InteropTransactionData.java (hunk 1 above)
Comment (josehernandezfintecheandomx) [reply fineract-provider/src/main/java/org/apache/fineract/interoperation/data/InteropTransactionData.java]: Done! the code has been updated to the service `InteropServiceImpl`

**Comment by josehernandezfintecheandomx** [comment]
> @josehernandezfintecheandomx Please find a way to fetch the Notes into the InterOpData as without it, it is a breaking change.
> 
> Please also check the spotless failures!

Done both things! @adamsaghy

File: fineract-provider/src/main/java/org/apache/fineract/interoperation/service/InteropServiceImpl.java [hunk 2]
```diff
@@ -218,7 +219,29 @@ public InteropTransactionsData getAccountTransactions(@NotNull String accountId,
             return (transactionsTo == null || transactionsTo.compareTo(transactionDate) > 0) && (transactionsFrom == null
                     || transactionsFrom.compareTo(transactionDate.withHour(23).withMinute(59).withSecond(59)) <= 0);
         };
-        return InteropTransactionsData.build(savingsAccount, transFilter);
+        InteropTransactionsData interopTransactionsData = InteropTransactionsData.build(savingsAccount, transFilter);
+        for (InteropTransactionData interopTransactionData : interopTransactionsData.getTransactions()) {
+            final List<Note> transactionNotes = noteRepository
+                    .findBySavingsTransactionId(Long.valueOf(interopTransactionData.getTransactionId()));
+            StringBuilder sb = new StringBuilder();
+            if (!transactionNotes.isEmpty()) {
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/interoperation/service/InteropServiceImpl.java]: This if condition is unnecessary

**Review by adamsaghy — APPROVED** [review]
LGTM

**Comment by adamsaghy** [comment]
@josehernandezfintecheandomx Please check my comments and please modify the PR title and the commit message:

Fineract modularisation tracked in this story: https://issues.apache.org/jira/browse/FINERACT-1932 

Please use this in the PR and in the commit message!

**Review by adamsaghy — CHANGES_REQUESTED** [review]
@josehernandezfintecheandomx Please check my comments and please modify the PR title and the commit message:

Fineract modularisation tracked in this story: https://issues.apache.org/jira/browse/FINERACT-1932

Please use this in the PR and in the commit message!

File: fineract-provider/src/main/java/org/apache/fineract/interoperation/service/InteropServiceImpl.java (hunk 2 above)
Comment (josehernandezfintecheandomx) [reply fineract-provider/src/main/java/org/apache/fineract/interoperation/service/InteropServiceImpl.java]: Done! if removed

**Comment by josehernandezfintecheandomx** [comment]
> @josehernandezfintecheandomx Please check my comments and please modify the PR title and the commit message:
> 
> Fineract modularisation tracked in this story: https://issues.apache.org/jira/browse/FINERACT-1932
> 
> Please use this in the PR and in the commit message!

@adamsaghy Done both things requested, Thanks!

**Pull request merged**